# HG changeset patch
# User Benjamin Lindner <lindnerb@users.sourceforge.net>
# Date 1222968357 14400
#      Thu Oct 02 13:25:57 2008 -0400
# Node ID 9ba45b125ee8ac596cda6748a372297079d142f0
# Parent  15ffb9836c01675586133963fd91d55926678c1d
enclose building direcries in quotes in pkg.m

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2008-10-02  Benjamin Lindner <lindnerb@users.sourceforge.net>
+
+	* pkg.m (configure_make): Enclose building directory in quotes.
+
 2008-10-02  Ben Abbott <bpabbott@mac.com>
 
 	* plot/__go_draw_axes__.m: Remove depdenence on gnuplot version..
 
 2008-09-30  Jaroslav Hajek  <highegg@gmail.com>
 
 	* string/split.m: New tests.
 
diff --git a/scripts/pkg/pkg.m b/scripts/pkg/pkg.m
--- a/scripts/pkg/pkg.m
+++ b/scripts/pkg/pkg.m
@@ -1223,31 +1223,31 @@ function configure_make (desc, packdir, 
         flags = cstrcat (flags, " CXX=\"", octave_config_info ("CXX"), "\"");
       endif
       if (isempty (getenv ("AR")))
         flags = cstrcat (flags, " AR=\"", octave_config_info ("AR"), "\"");
       endif
       if (isempty (getenv ("RANLIB")))
         flags = cstrcat (flags, " RANLIB=\"", octave_config_info ("RANLIB"), "\"");
       endif
-      [status, output] = shell (cstrcat ("cd ", src, "; ./configure --prefix=\"",
+      [status, output] = shell (strcat ("cd '", src, "'; ./configure --prefix=\"",
                                         desc.dir, "\"", flags));
       if (status != 0)
 	rm_rf (desc.dir);
 	error ("the configure script returned the following error: %s", output);
       elseif (verbose)
 	printf("%s", output);
       endif
 
     endif
 
     ## make
     if (exist (fullfile (src, "Makefile"), "file"))
       [status, output] = shell (cstrcat ("export INSTALLDIR=\"", desc.dir,
-					 "\"; make -C ", src));
+					 "\"; make -C '", src, "'"));
       if (status != 0)
 	rm_rf (desc.dir);
 	error ("'make' returned the following error: %s", output);
       elseif (verbose)
 	printf("%s", output);
       endif
     endif
 
